### DISCUSSION TITLE
What description should I add for training Entity Linker?

### DISCUSSION THREAD

**Discussion by shrinidhin** [explosion/spaCy#10686] [Help: Coding & Implementations] [answered]
Hello!I have been working on building an entity linker for linking companies. I have list of unique company IDs along with their company name. I've been following the [ner-emerson tutorial](https://github.com/explosion/projects/tree/v3/tutorials/nel_emerson).
I need help with two things:
1. I see that you need to add description texts in order to generate the unique entity vectors. Won't the company names (Which are all unique in my Knowledge Base) be enough to be used as entity vectors?Is adding a description text and using the vector embedding mandatory? I read the following [stackoverflow discussion](https://stackoverflow.com/questions/59882071/spacy-entity-linking-word-vectors) which says that "a description of an entity will semantically be similar to the sentence/context in which an entity is used.". so in this case what will I have to add in the description?Do I need to add example sentences or do I need to add a description of the company? 
2. What values need to be added to the `entity_vector_length` parameter while creating a `KnowledgeBase` in spacy? I gave 300 as the input as mentioned and it gives me an error with the message: `ValueError: [E141] Entity vectors should be of length 300 instead of the provided 96.`
Any inputs would be really helpful. thank you!

**Comment by svlandeg** [comment, marked as answer]
Hi!

The entity linker currently does expect descriptions that are semantically meaningful. The training algorithm will learn the similarity between the descriptions and the sentences in which entities appear. It was originally designed with something like Wikidata in mind, where you'd have short but meaningful descriptions. I guess you could try training this with company names as descriptions - but it might not work as well.

With respect to the vector length, you'll need to provide the correct dimension of your vectors. It sounds like your vectors are of length 96, so you'll have to give that as input.

**Reply by shrinidhin** [reply]
So what kind of text can be possibly added as a description?For my use case, the entities to be identified are names of stocks in news articles. So will adding sample example sentences from the news articles help?I am currently adding the name of the company along with the industry the company belongs to (Say Real Estate, Technology etc).

**Reply by svlandeg** [reply]
Let me phrase that question differently: if you have two names of stocks that are the same, which differentiating information would you as a human use to find out which of the two entries in the KB it is? If it's the company name, and that is typically also available in the sentence context, then yes I definitely think that approach could work.

**Reply by shrinidhin** [reply]
Ok. I understand it now. Thank you so much!
